rfcs/text/0018_timeslider.md

```diff
@@ -0,0 +1,217 @@
+  timeslice?: Timeslice; // The currently displayed timeslice. Needs to be set after onTimesliceChange to be reflected back in UI. If ommitted, the control selects the first timeslice.
+}
+```
+
+`timeslice` is clamped to the bounds of `timeRange`.
+
+Any change to `timeslice`, either by dragging the handles of the timeslider, or pressing the back or forward buttons, calls the `onTimesliceChanged` handler.
+
+Since the initial use is inside Maps, the initial location of this React-component is inside the Maps plugin, `x-pack/plugins/maps/public/timeslider`.
+
+Nonetheless, this UI-component should be easily "copy-pastable" to another location.
+
+### 2.2.2 Internals
+
+The timeslider automatically subdivides the timerange with equal breaks that are heuristically determined. 
+
+It assigns about 6-10 breaks within the `timerange`, snaps the "ticks" to a natural "pretty date" using `calcAutoIntervalNear` from `data/common`. 
+
+For example;
+- a `timerange` of 8.5 days, it would assign a 8 day-steps, plus some padding on either end, depending on the entire `timerange`.
+- a `timerange` of 6.9 years would snap to year-long step, plus some padding on either end, depending on the entire `timerange`.
+
+The slider itself is a `<EuiDualRange>`.
+
+### 2.2.2 End-user behavior
+
+- the user can manipulate the `timeslice`-double ranged slider to any arbitrary range within `timerange`.
+- the user can press the forward and back buttons for a stepped navigation. The range of the current time-slice is preserved when there is room for the `timeslice` within the `timerange`. 
+     - when the user has _not modified the width_ of the `timeslice`, using the buttons means stepping through the pre-determined ticks (e.g. by year, by day, ...)
+     - when the user has _already modified the width_ of the `timeslice`, it means stepping through the `timerange`, with a stride of the width of the `timeslice`.
+- the `timeslice` "snaps" to the beginning or end (depending on direction) of the `timerange`. In practice, this means the `timeslice` will collaps or reduce in width.
+    
+## 2.3 Maps integration of the timeslider React component
+
+This control will be integrated in the Maps-UI.
+
+Maps is Redux-based, so `timeslice` selection and activation/de-activation all propagates to the Redux-store.
+
+#### 2.3.1 Position in the UI
+
+The timeslider control is enabled/disabled by the timeslider on/off toggle-button in the main toolbar.
+
+![Timeslider version 1](../images/timeslider/toolbar.png)
+
+
+#### 2.3.2 Layer interactions
+
+
+Enabling the Timeslider will automatically retrigger refresh of _all_ time-based layers to the currently selected `timeslice`.
+
+The Layer-TOC will indicate which layer is currently "time-filtered" by the timeslider.
+
+On a layer-per-layer basis, users will be able to explicitly opt-out if they should be governed by the timeslider or not. 
```

Comment:
Question: should this be timeslider or timerange? The layer can turn of global time filters. I assume this would also disable timeslider and there is not a separate flag for each. May be this should clarify that each layer can turn of time filtering and state that means either by timerange or timeslice